src/pudl/etl/glue_assets.py

```diff
@@ -389,11 +404,16 @@ def augement_crosswalk_with_bga_eia860(
     """Merge all EIA Unit IDs into the crosswalk."""
     return crosswalk_clean.merge(
         boiler_generator_assn_eia860[
-            ["plant_id_eia", "generator_id", "unit_id_pudl"]
+            [
+                "plant_id_eia",
+                "generator_id",
+                "unit_id_pudl",
+                "boiler_id",
+            ]
         ].drop_duplicates(),
         how="outer",
-        on=["plant_id_eia", "generator_id"],
-        validate="m:1",
+        on=["plant_id_eia", "generator_id", "boiler_id"],
+        validate="m:m",
     )
```

Comment:
since this merge is happening pre-network x, it seems like we should probably revert these changes no? so there is no boiler_id pre-network_x.

Reply:
I don't think it really matters? Because we need to merge with the bga table for the `unit_id` which is used in the network x process. We could drop the boiler_id more explicitly before the process, but it wouldn't make a huge difference.